src/app/context.js

```diff
@@ -299,6 +302,33 @@ export default class Context {
         });
     }
 
+    /**
```

Comment:
@will-moore I know there is some setup code in context.js, the big difference to the open with is that this code needs an absolute guarantee to run before anything else, specifically the aurelia initialization which I have pretty much no control over in terms of order.

I guess what I'm saying is that I'd rather have the bulk of the code out of context and closer to where its happening. I try to encapsulate the functionality right in the files where it's needed. If there is a needed sequence of actions I think it should be achievable via callbacks in your case.

As a matter of fact, there is enough code and most of it being self contained/global in nature that I'd put it into its own file and then just call with the few parameters needed in the respective places.

I think the utils directory might be a good place and you could do it as in https://github.com/ome/omero-iviewer/blob/master/src/utils/misc.js with "static" methods so that you just need to call something like ``` OpenWith.init ```

Reply:
forgot to mention: the noView decorator would be needed for that additional file (like Misc class), anything else should be more or less analogous (export and import statements).

Reply:
It's kinda analogous to the ```setUpLuts()``` in that we need to load the open-with options ASAP. In fact, we also need to load the ```openwith.js``` file for each open-with and these need to be loaded before the right-panel renders.
When testing with local server (single request processed at a time) I sometimes found that the right panel Info tab would render before the ```openwith.js``` loaded e.g. for webgateway viewer. This would then give an incorrect URL to that viewer.
I didn't see this problem when using nginx - since the simultaneous processing of request meant that the ```openwith.js``` was generally returned faster - before the ```imgData``` json and before the Info tab rendered. But this is still not guaranteed.

I'll move as much of the code as I can to it's own file.

Reply:
Yes. Moving as much as you can is a good idea.

On an aside, setUpLuts is instantly needed because the settings options display the png instantly, the actual height depending on the number of luts is needed and image data uses the luts table for lookup table internally. Settings are also the default to show. T

Open with would still render correctly (and is waiting on image_info ready already) if the script pulling wasn't done right away, right? Another idea is a loaded counter in a global/context variable and the open with links being enabled once everything is there. 

As a matter of fact, perhaps what I said in the above paragraph is not even necessary because the open_with_links array that you use should synchronize between ui and collection that is to say once you add it to the array, the ui will update. So before you add an entry, the link will not be there even.